packages/core/src/modules/discover-features/DiscoverFeaturesEvents.ts

```diff
@@ -0,0 +1,28 @@
+import type { BaseEvent } from '../../agent/Events'
+import type { ConnectionRecord } from '../connections'
+import type { Feature, FeatureQueryOptions } from './models'
+
+export enum DiscoverFeaturesEventTypes {
+  QueryReceived = 'QueryReceived',
+  DisclosureReceived = 'DisclosureReceived',
+}
+
+export interface DiscoverFeaturesQueryReceivedEvent extends BaseEvent {
+  type: typeof DiscoverFeaturesEventTypes.QueryReceived
+  payload: {
+    queries: FeatureQueryOptions[]
+    protocolVersion: string
+    connection: ConnectionRecord
+    threadId: string
+  }
+}
+
+export interface DiscoverFeaturesDisclosureReceivedEvent extends BaseEvent {
+  type: typeof DiscoverFeaturesEventTypes.DisclosureReceived
+  payload: {
+    disclosures: Feature[]
```

Comment:
Should we include the message in both events? If people have custom decorators for example it may be useful to expose those for extensibility